### PR TITLE
reprepro 5.4.4

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/r/reprepro.rb [hunk 1]
```diff
@@ -36,16 +36,20 @@ class Reprepro < Formula
     cause "No support for GNU C nested functions"
   end
 
+  # Replace strcmp2 with strcmp for consistent string comparison
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/5426d605c0d6b6bf4c2e315fed7a991f447caf1a/reprepro/5.4.4-strcmp2.patch"
+    sha256 "cf1aacf08960cb89896a739086d170417820fe894044d0f446b4ff303a1182e5"
+  end
```
Comment (carlocab) [inline Formula/r/reprepro.rb:43]: Has this been upstreamed?

Comment (chenrui333) [reply Formula/r/reprepro.rb:43]: not really, I can push it now.

Comment (chenrui333) [reply Formula/r/reprepro.rb]: https://salsa.debian.org/debian/reprepro/-/merge_requests/10

File: Formula/r/reprepro.rb [hunk 2]
```diff
@@ -36,16 +36,21 @@ class Reprepro < Formula
     cause "No support for GNU C nested functions"
   end
 
+  # Replace strcmp2 with strcmp for consistent string comparison
+  # upstream patch PR, https://salsa.debian.org/debian/reprepro/-/merge_requests/10
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/5426d605c0d6b6bf4c2e315fed7a991f447caf1a/reprepro/5.4.4-strcmp2.patch"
+    sha256 "cf1aacf08960cb89896a739086d170417820fe894044d0f446b4ff303a1182e5"
+  end
```
Comment (carlocab) [inline Formula/r/reprepro.rb:44]: I don't think the patch is correct; it seems likely to result in UB: https://salsa.debian.org/debian/reprepro/-/blob/21a227f5fb380672ba022d47628895d2f023bde9/globals.h#L97

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/r/reprepro.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/r/reprepro.rb:44]: yeah, I thought the same thing, I will debug a bit today.

Comment (carlocab) [reply Formula/r/reprepro.rb:44]: Maybe it's just missing an
```
include "globals.h"
```
somewhere?

**Pull request closed without merging**